vividus-plugin-mobitru/src/main/java/org/vividus/mobitru/selenium/MobitruCapabilitiesAdjuster.java

```diff
@@ -46,6 +51,7 @@ public Map<String, Object> getExtraCapabilities(DesiredCapabilities desiredCapab
         {
             deviceId = mobitruFacade.takeDevice(desiredCapabilities);
             mobitruFacade.installApp(deviceId, appFileName, installApplicationOptions);
+            testContext.put(KEY, deviceId);
```

Comment:
I guess device ID should be available in session capabilities, thus I would avoid managing this data on test data

Reply:
I've checked and it's not avaialble

Reply:
1. Web iOS - capabilities from server:
    ```json
    {
      "appium:dc" : "US",
      "appium:networkConnectionEnabled" : false,
      "appium:takesScreenshot" : true,
      "appium:automationName" : "XCUITest",
      "appium:locationContextEnabled" : false,
      "appium:platformVersion" : "17.3",
      "appium:webDriverAgentUrl" : "http://mstf-node:7504",
      "appium:databaseEnabled" : false,
      "appium:udid" : "00008130-001958513EBA001C",
      "browserName" : "Safari",
      "appium:webStorageEnabled" : false,
      "appium:deviceName" : "IPHONE iPhone16,1",
      "appium:javascriptEnabled" : true,
      "platformName" : "IOS"
    }
    ```
    I guess, it is `appium:udid`, isn't it?
2. Web Android - capabilities from server:
    ```json
    {
      "appium:dc" : "US",
      "appium:chromedriverPort" : 37415,
      "appium:takesScreenshot" : true,
      "appium:warnings" : { },
      "appium:desired" : {
        "browserName" : "Chrome",
        "goog:chromeOptions" : {
          "args" : [ ],
          "extensions" : [ ]
        },
        "platformName" : "Android",
        "platformVersion" : "14",
        "deviceName" : "SAMSUNG SM-S928U1",
        "udid" : "127.0.0.1:37293",
        "dc" : "US",
        "chromedriverExecutableDir" : "/mount/chromedriver",
        "chromedriverPort" : 37415
      },
      "appium:deviceApiLevel" : 34,
      "appium:locationContextEnabled" : false,
      "appium:deviceScreenSize" : "1080x2340",
      "appium:deviceManufacturer" : "samsung",
      "appium:chromedriverExecutableDir" : "/mount/chromedriver",
      "appium:udid" : "127.0.0.1:37293",
      "appium:pixelRatio" : 2.8125,
      "goog:chromeOptions" : {
        "args" : [ ],
        "extensions" : [ ]
      },
      "browserName" : "Chrome",
      "platformName" : "ANDROID",
      "appium:networkConnectionEnabled" : true,
      "appium:deviceScreenDensity" : 450,
      "appium:viewportRect" : {
        "left" : 0,
        "top" : 76,
        "width" : 1080,
        "height" : 2146
      },
      "appium:deviceModel" : "SM-S928U1",
      "appium:platformVersion" : "14",
      "appium:databaseEnabled" : false,
      "appium:deviceUDID" : "127.0.0.1:37293",
      "appium:statBarHeight" : 76,
      "appium:webStorageEnabled" : false,
      "appium:appActivity" : "com.google.android.apps.chrome.Main",
      "appium:deviceName" : "SAMSUNG SM-S928U1",
      "appium:javascriptEnabled" : true,
      "appium:appPackage" : "com.android.chrome"
    }
    ```
    `appium:deviceUDID` looks incorrect, is it Mobitru bug?
3. Native mobile iOS app - capabilities from server:
    ```json
    {
      "appium:dc" : "CIS",
      "appium:networkConnectionEnabled" : false,
      "mobitru-device-search:manufacturer" : "IPAD",
      "appium:takesScreenshot" : true,
      "appium:automationName" : "XCUITest",
      "appium:locationContextEnabled" : false,
      "appium:platformVersion" : "17.2",
      "appium:webDriverAgentUrl" : "http://mstf-node:7480",
      "mobitru-device-search:model" : "iPad",
      "appium:databaseEnabled" : false,
      "appium:udid" : "00008020-000669981A43002E",
      "appium:webStorageEnabled" : false,
      "appium:deviceName" : "IPAD iPad11,1",
      "appium:javascriptEnabled" : true,
      "platformName" : "IOS",
      "mobitru-device-search:type" : "tablet"
    }
    ```
    I guess, it is `appium:udid`, isn't it?
4. Native Android app - capabilities from server:
    ```json
    {
      "appium:dc" : "EU",
      "appium:chromedriverPort" : 45497,
      "appium:takesScreenshot" : true,
      "appium:warnings" : { },
      "appium:desired" : {
        "mobitru-device-search:type" : "phone",
        "platformName" : "Android",
        "mobitru-device-search:manufacturer" : "SAMSUNG",
        "udid" : "127.0.0.1:42833",
        "automationName" : "UIAutomator2",
        "platformVersion" : "12",
        "deviceName" : "SAMSUNG SM-A525F",
        "dc" : "EU",
        "chromedriverExecutableDir" : "/mount/chromedriver",
        "chromedriverPort" : 45497
      },
      "appium:deviceApiLevel" : 31,
      "appium:automationName" : "UIAutomator2",
      "appium:locationContextEnabled" : false,
      "appium:deviceScreenSize" : "1080x2400",
      "appium:deviceManufacturer" : "samsung",
      "appium:chromedriverExecutableDir" : "/mount/chromedriver",
      "appium:udid" : "127.0.0.1:42833",
      "appium:pixelRatio" : 2.625,
      "platformName" : "ANDROID",
      "mobitru-device-search:type" : "phone",
      "appium:networkConnectionEnabled" : true,
      "mobitru-device-search:manufacturer" : "SAMSUNG",
      "appium:deviceScreenDensity" : 420,
      "appium:viewportRect" : {
        "left" : 0,
        "top" : 87,
        "width" : 1080,
        "height" : 2099
      },
      "appium:deviceModel" : "SM-A525F",
      "appium:platformVersion" : "12",
      "appium:databaseEnabled" : false,
      "appium:deviceUDID" : "127.0.0.1:42833",
      "appium:statBarHeight" : 87,
      "appium:webStorageEnabled" : false,
      "appium:deviceName" : "SAMSUNG SM-A525F",
      "appium:javascriptEnabled" : true
    }
    ```
    the same as for web Android

Reply:
yes and please look at p.4 where there is no real UDID of the Device:
`
"udid" : "127.0.0.1:42833",
"appium:udid" : "127.0.0.1:42833",
"appium:deviceUDID" : "127.0.0.1:42833",
`

only IPs

Reply:
so obviously it's a Mobitru bug, I believe many people are affected, is this bug going to be fixed?

Reply:
this is not a bug, because this option is not required by official Appium capabilities and, according to this, this situation doesn't affect our users

Reply:
it's a bug because `127.0.0.1:42833` is not UDID, if this option is not mandatory, then it could be removed, but as it's present, it contains invalid value

Reply:
the bug has been reported to Support, nevertheless, there should be some class which represents Mobitru storage, e.g. `MobitruTestContext`, so that data sharing won't happen by re-using key and access test context

Reply:
it's not a bug :) because ip:port as UDID is the normal behaviour for Android adb connection **over network** (not usb):

more details can be found here https://developer.android.com/tools/adb#:~:text=Confirm%20that%20your%20host%20computer%20is%20connected%20to%20the%20target%20device%3A